src/java.desktop/windows/classes/sun/awt/windows/WFramePeer.java

```diff
@@ -80,11 +79,9 @@ public void toFront() {
     private native void setMaximizedBounds(int x, int y, int w, int h);
     private native void clearMaximizedBounds();
 
-    @SuppressWarnings("removal")
     private static final boolean keepOnMinimize = "true".equals(
-        AccessController.doPrivileged(
             new GetPropertyAction(
-            "sun.awt.keepWorkingSetOnMinimize")));
+            "sun.awt.keepWorkingSetOnMinimize"));
```

Comment:
Shouldn't this be changed to `System.getProperty`? `GetPropertyAction` implements `PrivilegedAction` and is deprecated for removal (so I'm surprised that this doesn't generate a fatal warning).

Reply:
Even more to the point: this won't work the way it used to: the former call to `doPrivileged` returned a String (the value of `GetPropertyAction::run`). The new code will use the `GetPropertyAction` instance itself without executing the run method, so you will be comparing a string `"true"` to the new `GetPropertyAction` instance, which will always be false since they aren't of the same type, regardless of what you set the property to. We had a similar issue with one of our JavaFX PRs that was only discovered by testing.

Reply:
Yes, this is not "RUNNING" the action so it isn't right.
And we should change it to a simple System.getProperty as Kevin points out.

Reply:
yes, this was missed..updated..